integrations/Podsights/browser.js

```diff
@@ -0,0 +1,159 @@
+/* eslint-disable no-underscore-dangle */
+/* eslint-disable class-methods-use-this */
+import get from "get-value";
+import sha256 from "crypto-js/sha256";
+import {
+  ADD_TO_CART_EVENT,
+  CHECK_OUT_EVENT,
+  LEAD_EVENT,
+  NAME,
+  PRODUCT_EVENT,
+  PURCHASE_EVENT,
+} from "./constants";
+import ScriptLoader from "../ScriptLoader";
+import logger from "../../utils/logUtil";
+import {
+  getHashFromArrayWithDuplicate,
+  removeUndefinedAndNullValues,
+} from "../utils/commonUtils";
+import { constructPayload } from "../../utils/utils";
+import { payloadBuilder, payloadBuilderInList } from "./utils";
+
+class Podsights {
+  constructor(config, analytics) {
+    this.pixelId = config.pixelId;
+    this.eventsToPodsightsEvents = config.eventsToPodsightsEvents;
+    this.enableAliasCall = config.enableAliasCall;
+    this.name = NAME;
+    if (analytics.logLevel) logger.setLogLevel(analytics.logLevel);
+  }
+
+  init() {
+    logger.debug("===In init Podsights===");
+    window.pdst =
+      window.pdst ||
+      function () {
+        (window.pdst.q = window.pdst.q || []).push(arguments);
+      };
+    ScriptLoader("pdst-capture", "https://cdn.pdst.fm/ping.min.js");
+    window.pdst("conf", { key: `${this.pixelId}` });
+  }
+
+  isLoaded() {
+    logger.debug("===In isLoaded Podsights===");
+    return !!(window.pdst && typeof window.pdst === "function");
+  }
+
+  isReady() {
+    logger.debug("===In isReady Podsights===");
+    return !!(window.pdst && typeof window.pdst === "function");
+  }
+
+  loadAliasEvent(externalId) {
+    if (this.enableAliasCall && externalId) {
+      window.pdst("alias", {
+        id: sha256(externalId).toString(),
+      });
+    }
+  }
+
+  /**
+   * Track - tracks an event for an user
+   * @param {Track} track
+   */
+  track(rudderElement) {
+    logger.debug("===In Podsights Track===");
+    const { message } = rudderElement;
+    const { event, properties } = message;
+    if (!event) {
+      logger.error("[Podsights]: event name from track call is missing.");
+      return;
+    }
+
+    const eventsMapping = getHashFromArrayWithDuplicate(
```

Comment:
let us also add some default mapping. The priority will be given to user defined mapping, and if not found there we can fallback to our default mapping. if that is also not matching then throw error

Reply:
Does default mapping mean with RudderStack standard event?

Reply:
done

Reply:
lets add this in research spec and let the doc team know